File: LearningManagementBackend/src/main/java/org/learn/learningmanagementbackend/model/Semester.java
package org.learn.learningmanagementbackend.model;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDate;
import java.util.ArrayList;
import java.util.List;

@Setter
@Getter
@Entity
@Table(name = "Semester")
public class Semester extends BaseEntity{

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "academic_year_id")
    private AcademicYear academicYear;

    @Column(name = "name")
    private String name;

    @Column(name = "start_date")
    private LocalDate startDate;

    @Column(name = "end_date")
    private LocalDate endDate;

    @Column(name = "is_active")
    private Boolean isActive;

    @OneToMany(mappedBy = "semester", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<TuitionInvoice> tuitionInvoices = new ArrayList<>();

    public void addTuitionInvoice(TuitionInvoice tuitionInvoice){
        this.tuitionInvoices.add(tuitionInvoice);
        tuitionInvoice.setSemester(this);
    }

    @OneToMany(mappedBy = "semester", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<StudentSemesterSummary> studentSemesterSummaries = new ArrayList<>();

    public void addStudentSemesterSummary(StudentSemesterSummary studentSemesterSummary){
        this.studentSemesterSummaries.add(studentSemesterSummary);
        studentSemesterSummary.setSemester(this);
    }
}

File: learning_management_app/lib/models/admin/semester.dart
class Semester {
  final int? id;
  final int? academicYearId;
  final String? academicYearName;
  final String? name;
  final String? startDate;
  final String? endDate;
  final bool? isActive;

  Semester({
    this.id,
    this.academicYearId,
    this.academicYearName,
    this.name,
    this.startDate,
    this.endDate,
    this.isActive,
  });

  factory Semester.fromJson(Map<String, dynamic> json) {
    return Semester(
      id: json['id'],
      academicYearId: json['academicYearId'],
      academicYearName: json['academicYearName'],
      name: json['name'],
      startDate: json['startDate'],
      endDate: json['endDate'],
      isActive: json['isActive'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'academicYearId': academicYearId,
      'name': name,
      'startDate': startDate,
      'endDate': endDate,
      'isActive': isActive,
    };
  }
}
